app/lib/lib/roomTypes/newsfeed.js

```diff
@@ -0,0 +1,128 @@
+import { Meteor } from 'meteor/meteor';
+
+import { openRoom } from '../../../ui-utils';
+import { ChatRoom, ChatSubscription } from '../../../models';
+import { settings } from '../../../settings';
+import { hasAtLeastOnePermission } from '../../../authorization';
+import { getUserPreference, RoomTypeConfig, RoomTypeRouteConfig, RoomSettingsEnum, UiTextContext } from '../../../utils';
+import { getAvatarURL } from '../../../utils/lib/getAvatarURL';
+
+export class NewsfeedRoomRoute extends RoomTypeRouteConfig {
+	constructor() {
+		super({
+			name: 'newsfeed',
+			path: '/newsfeed/:name',
+		});
+	}
+
+	action(params) {
+		return openRoom('n', params.name);
+	}
+}
+
+export class NewsfeedRoomType extends RoomTypeConfig {
+	constructor() {
+		super({
+			identifier: 'n',
+			order: 30,
+			icon: 'star',
+			label: 'Newsfeed',
+			route: new NewsfeedRoomRoute(),
+		});
+	}
+
+	getIcon(roomData) {
+		if (roomData.prid) {
+			return 'discussion';
+		}
+		return this.icon;
+	}
+
+	findRoom(identifier) {
+		const query = {
+			t: 'n',
+			name: identifier,
+		};
+		return ChatRoom.findOne(query);
+	}
+
+	roomName(roomData) {
+		if (roomData.prid) {
+			return roomData.fname;
+		}
+		if (settings.get('UI_Allow_room_names_with_special_chars')) {
+			return roomData.fname || roomData.name;
+		}
+		return roomData.name;
+	}
+
+	condition() {
+		const groupByType = getUserPreference(Meteor.userId(), 'sidebarGroupByType');
+		return groupByType && (hasAtLeastOnePermission(['view-c-room', 'view-joined-room']) || settings.get('Accounts_AllowAnonymousRead') === true);
+	}
+
+	showJoinLink(roomId) {
+		return !!ChatRoom.findOne({ _id: roomId, t: 'n' });
+	}
+
+	includeInRoomSearch() {
+		return true;
+	}
+
+	isGroupChat() {
+		return true;
+	}
+
+	canAddUser(room) {
+		return hasAtLeastOnePermission(['add-user-to-any-c-room', 'add-user-to-joined-room'], room._id);
+	}
+
+	canSendMessage(roomId) {
+		const room = ChatRoom.findOne({ _id: roomId, t: 'n' }, { fields: { prid: 1 } });
```

Comment:
For what type of user is this check implemented?
If I understand correctly, messages in newsfeed type channel are only populated via the server. What type of user will access this method?

Reply:
Again for the same reason as above. I will remove these only relaying is completely implemeted